ddtrace/sampler.py

```diff
@@ -161,6 +163,10 @@ def sample(self, span):
 
         span.set_metric(SAMPLING_AGENT_DECISION, sampler.sample_rate)
         self._set_priority(span, priority)
+
+        sampling_mechanism = SamplingMechanism.DEFAULT if key == self._default_key else SamplingMechanism.AGENT_RATE
```

Comment:
We only get the rate by service values after making a request to the trace agent.

So we will also use the default for the first payload.

It can also happen any time we see a new service name that we haven't seen before, we will only have a key/rate for it from the trace agent after the first payload including that trace/service.

Reply:
@majorgreys can this be resolved?

Reply:
Yes, we are checking the key each time, consistent with the current behavior.